File: opcua-webservice-spring/examples/uawebservicedemo/src/main/java/org/opcfoundation/uawebservicedemo/models/employee/employeeskill/EmployeeSkillReferenceType.java
package org.opcfoundation.uawebservicedemo.models.employee.employeeskill;

import org.eclipse.milo.opcua.stack.core.StatusCodes;
import org.eclipse.milo.opcua.stack.core.UaRuntimeException;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.uawebservicedemo.database.EmployeeDatabase;
import org.opcfoundation.uawebservicedemo.database.entity.Skill;
import org.opcfoundation.uawebservicedemo.database.mapper.SkillMapper;
import org.opcfoundation.uawebservicedemo.models.EmployeeTwinSpace;
import org.opcfoundation.uawebservicedemo.models.company.skillclassification.SkillClassType;
import org.opcfoundation.webserver.digitaltwin.element.ReferenceElementType;
import org.opcfoundation.webserver.service.message.digitaltwin.GetLinkRequest;
import org.opcfoundation.webserver.service.message.digitaltwin.GetLinkResponse;
import org.opcfoundation.webserver.types.digitaltwin.ReferenceTargetDescriptor;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public class EmployeeSkillReferenceType extends ReferenceElementType {
    public EmployeeSkillReferenceType(EmployeeTwinSpace twinSpace) {
        super("EmployeeSkillReferenceType", new LocalizedText("EmployeeSkillReference"), twinSpace);
    }

    public CompletableFuture<GetLinkResponse> onGetLinks(GetLinkRequest request)
    {
        try
        {
            // Get employee id from request
            Integer employeeId = Integer.parseInt(request.getId());

            // Database query
            SkillMapper skillMapper = EmployeeDatabase.getSkillMapper();
            List<Skill> skills = skillMapper.getSkillOfEmployee(employeeId);

            // Return general information of child department
            GetLinkResponse response = new GetLinkResponse();
            SkillClassType skillClassType = ((EmployeeTwinSpace)nodeManager).skillClassType;

            for (Skill item : skills)
            {
                ReferenceTargetDescriptor descriptor = new ReferenceTargetDescriptor(
                        item.getId().toString(), // Child id
                        new LocalizedText(item.getName()), // Child name
                        skillClassType); // Child object type

                response.add(descriptor);
            }

            return CompletableFuture.completedFuture(response);
        } catch (Exception e) {
            throw new UaRuntimeException(StatusCodes.Bad_NodeIdUnknown);
        }
    }
}

File: opcua-webservice-node/examples/demoserver/src/models/employee/employeeskill/EmployeeSkillReferenceType.ts
import { StatusCodes } from "opcua-webapi";
import { UaError, UaLocalizedText } from "opcua-webapi-ts";
import {
    GetLinkRequest,
    GetLinkResponse,
    ReferenceElementType,
    ReferenceTargetDescriptor
} from "opcua-webservice-node";

import { prisma } from "../../../connectors/prismaClient";
import { EmployeeTwinSpace } from "../../EmployeeTwinSpace";

export class EmployeeSkillReferenceType extends ReferenceElementType {
    constructor(twinSpace: EmployeeTwinSpace) {
        super("EmployeeSkillReferenceType", new UaLocalizedText("EmployeeSkillReference"), twinSpace);
    }

    override async onGetLinks(request: GetLinkRequest): Promise<GetLinkResponse>
    {
        // Get employee id from request
        const employeeId = parseInt(request.id);
        if (isNaN(employeeId)) throw UaError.from(StatusCodes.BadInvalidArgument);
        
        // Database query
        const employee = await prisma.employee.findUnique({
            where: { ID: employeeId },
            select: {
                employeeskill: {
                    select: {
                        skill: { select: { ID: true, SkillName: true } }
                    }                    
                }
                }
            });

        if (!employee) throw UaError.from(StatusCodes.BadNodeIdUnknown);

        // Return general information of child department
        let response = new GetLinkResponse();

        for (const item of employee.employeeskill) {
            let current = item.skill;

            const descriptor = new ReferenceTargetDescriptor(
                current.ID.toString(), // Child id
                new UaLocalizedText(current.SkillName), // Child name
                EmployeeTwinSpace.SkillClassType);

            response.add(descriptor);
        }

        return response;
    }
}